Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -168,6 +220,175 @@ namespace PhysX
         return AZ::Edit::PropertyRefreshLevels::None;
     }
 
+    AZ::u32 EditorHeightfieldColliderComponent::SaveHeightfieldAssetToDisk()
+    {
+        auto assetType = AZ::AzTypeInfo<PhysX::Pipeline::HeightFieldAsset>::Uuid();
+        auto assetHandler = const_cast<AZ::Data::AssetHandler*>(AZ::Data::AssetManager::Instance().GetHandler(assetType));
+        if (assetHandler)
+        {
+            char projectPath[AZ_MAX_PATH_LEN];
+            AZ::IO::FileIOBase::GetInstance()->ResolvePath("@projectroot@", projectPath, AZ_MAX_PATH_LEN);
+
+            AZStd::string heightfieldFullPath;
+            AzFramework::StringFunc::Path::ConstructFull(projectPath, m_bakedHeightfieldRelativePath.c_str(), heightfieldFullPath, true);
+
+            AZ::IO::FileIOStream fileStream(heightfieldFullPath.c_str(), AZ::IO::OpenMode::ModeWrite);
+            if (fileStream.IsOpen())
```

Comment:
tried it, but seems AzFramework::StringFunc::Path::ConstructFull doesn't resolve the aliases and we use it in all 3 places, so have to resolve the path first.

Reply:
It's submitted now, but your for future reference, you can avoid creating the path at all if the file is opened with the flag `AZ::IO::OpenMode::CreatePath`, it can be ORed to `ModeWrite`.